infrastructures/infrastructure-azure/src/test/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructureTest.java

```diff
@@ -0,0 +1,353 @@
+import org.ow2.proactive.resourcemanager.nodesource.NodeSource;
+import org.python.google.common.collect.Sets;
+
+
+public class AzureInfrastructureTest {
+
+    private AzureInfrastructure azureInfrastructure;
+
+    @Mock
+    private ConnectorIaasController connectorIaasController;
+
+    @Mock
+    private NodeSource nodeSource;
+
+    @Mock
+    private Node node;
+
+    @Mock
+    private ProActiveRuntime proActiveRuntime;
+
+    @Mock
+    private NodeInformation nodeInformation;
+
+    @Before
+    public void init() {
+        MockitoAnnotations.initMocks(this);
+        azureInfrastructure = new AzureInfrastructure();
+
+    }
+
+    @Test
+    public void testInitialParamateres() {
+        assertThat(azureInfrastructure.clientId, is(nullValue()));
+        assertThat(azureInfrastructure.secret, is(nullValue()));
+        assertThat(azureInfrastructure.domain, is(nullValue()));
+        assertThat(azureInfrastructure.subscriptionId, is(nullValue()));
+        assertThat(azureInfrastructure.authenticationEndpoint, is(nullValue()));
+        assertThat(azureInfrastructure.managementEndpoint, is(nullValue()));
+        assertThat(azureInfrastructure.resourceManagerEndpoint, is(nullValue()));
+        assertThat(azureInfrastructure.graphEndpoint, is(nullValue()));
+        assertThat(azureInfrastructure.rmHostname, is(not(nullValue())));
+        assertThat(azureInfrastructure.connectorIaasURL,
+                   is("http://" + azureInfrastructure.rmHostname + ":8080/connector-iaas"));
+        assertThat(azureInfrastructure.image, is(nullValue()));
+        assertThat(azureInfrastructure.vmSizeType, is(nullValue()));
+        assertThat(azureInfrastructure.vmUsername, is(nullValue()));
+        assertThat(azureInfrastructure.vmPassword, is(nullValue()));
+        assertThat(azureInfrastructure.vmPublicKey, is(nullValue()));
+        assertThat(azureInfrastructure.resourceGroup, is(nullValue()));
+        assertThat(azureInfrastructure.region, is(nullValue()));
+        assertThat(azureInfrastructure.numberOfInstances, is(1));
+        assertThat(azureInfrastructure.numberOfNodesPerInstance, is(1));
+        if (System.getProperty("os.name").contains("Windows")) {
+            assertThat(azureInfrastructure.downloadCommand,
+                       is("powershell -command \"& { (New-Object Net.WebClient).DownloadFile('http://" +
+                          azureInfrastructure.rmHostname + ":8080/rest/node.jar', 'node.jar') }\""));
+        } else {
+            assertThat(azureInfrastructure.downloadCommand,
+                       is("wget -nv http://" + azureInfrastructure.rmHostname + ":8080/rest/node.jar"));
+
+        }
+        assertThat(azureInfrastructure.privateNetworkCIDR, is(nullValue()));
+        assertThat(azureInfrastructure.staticPublicIP, is(true));
+        assertThat(azureInfrastructure.additionalProperties, is("-Dproactive.useIPaddress=true"));
+
+    }
+
+    @Test
+    public void testConfigure() {
```

Comment:
renamed to `testConfigureDoNotThrowIllegalArgumentExceptionWithValidParameters` in 06b250351a3cac447f5c0b0272c314c5bd90a0cb